nox/virtualenv.py

```diff
@@ -278,7 +278,7 @@ def is_offline() -> bool:
             # DNS resolution to detect situation (1) or (2).
             host = gethostbyname("repo.anaconda.com")
             return host is None
-        except:  # pragma: no cover # noqa E722
+        except BaseException:  # pragma: no cover
```

Comment:
Do we _really_ need `BaseException` here? I just kept from changing anything. Wouldn't `Exception` be better?

Reply:
On board with this too. As far as I'm aware `BaseException` is something of an anti-pattern as it can catch `KeyboardInterrupt` among other internal excpetions

Reply:
AFAICS this was not discussed on #314. Possibly this allows the user to switch to offline mode by interrupting a hanging DNS query? OT: Funny that we have an `auto_offline` flag for noxfile authors, but no option for the user to specify offline mode.